### PR TITLE
Bug 1770724: Added filesystem and Network to project dashboard

### PR DESCRIPTION
**Pull request by cloudbehl**
Screenshot from 2019-11-11 14-32-29
Screenshot from 2019-11-11 14-32-22

Signed-off-by: cloudbehl <cloudbehl@gmail.com>

### CODE REVIEW

File: frontend/public/components/dashboard/project-dashboard/queries.ts [hunk 1]
```diff
@@ -18,6 +20,12 @@ const queries = {
   [ProjectQueries.POD_COUNT]: _.template(
     `count(kube_pod_info{namespace='<%= project %>'}) by (namespace)<%= duration %>`,
   ),
+  [ProjectQueries.FILESYSTEM_USAGE]: _.template(
+    `(sum(node_filesystem_size_bytes{namespace='<%= project %>'}) - sum(node_filesystem_free_bytes{namespace='<%= project %>'}))<%= duration %>`,
```
Comment (rawagner) [inline frontend/public/components/dashboard/project-dashboard/queries.ts]: I think that the query should look like
```
sum by(namespace) (pod:container_fs_usage_bytes:sum{container="",pod!="",namespace='<%= project %>'})<%= duration %>
```

File: frontend/public/components/dashboard/project-dashboard/queries.ts [hunk 2]
```diff
@@ -18,6 +20,12 @@ const queries = {
...
+  ),
+  [ProjectQueries.NETWORK_IN_OUT_USAGE]: _.template(
+    `((sum(instance:node_network_receive_bytes_excluding_lo:rate1m{namespace='<%= project %>'})) + (sum(instance:node_network_transmit_bytes_excluding_lo:rate1m{namespace='<%= project %>'})))<%= duration %>`,
```
Comment (rawagner) [inline frontend/public/components/dashboard/project-dashboard/queries.ts]: maybe like this ?

```
sum by(namespace) ((sum(rate(container_network_transmit_bytes_total{container="POD",pod!="",namespace='<%= project %>'}[5m]))) + (sum(rate(container_network_receive_bytes_total{container="POD",pod!="",namespace='<%= project %>'}[5m]))))<%= duration %>
```

File: frontend/public/components/dashboard/project-dashboard/utilization-card.tsx [hunk 3]
```diff
@@ -145,6 +164,22 @@ export const UtilizationCard = withDashboardResources(
               error={memoryError}
               TopConsumerPopover={memPopover}
             />
+            <UtilizationItem
+              title="Filesystem"
+              data={filesystemStats}
+              isLoading={!projectName || !filesystemUtilization}
+              humanizeValue={humanizeBinaryBytes}
+              query={queries[ProjectQueries.FILESYSTEM_USAGE]}
+              error={filesystemError}
```
Comment (rawagner) [inline frontend/public/components/dashboard/project-dashboard/utilization-card.tsx]: can you also add top consumers ?

**Comment by rawagner** [comment]
@kyoto ptal

**Comment by kyoto** [comment]
We are running these queries through the `prometheus-tenancy`, which takes the `namespace` as a GET param. Therefore, it should not be necessary to also specify the `namespace` in the query.

I see that all queries in this file specify `namespace` labels, but if they are just for the project dashboard, which is limited to a single namespace by using `prometheus-tenancy`, we shouldn't need any `{namespace=...}` or `by (namespace)` in the queries.

Or am I missing something?

**Comment by kyoto** [comment]
Not specific to this PR, but I notice that this file is using single quotes for label values, but we use double quotes in our PromQL by convention (e.g. `namespace="default"` rather than `namespace='default'`).

**Comment by kyoto** [comment]
Also, as discussed on https://github.com/openshift/console/pull/2585, these graphs should really be generated with data from the Prometheus API's `query_range` endpoint, not the `query` endpoint.

**Comment by rawagner** [comment]
`query_range` endpoint is addressed in https://github.com/openshift/console/pull/3303

**Comment by rawagner** [comment]
> We are running these queries through the `prometheus-tenancy`, which takes the `namespace` as a GET param. Therefore, it should not be necessary to also specify the `namespace` in the query.
> 
> I see that all queries in this file specify `namespace` labels, but if they are just for the project dashboard, which is limited to a single namespace by using `prometheus-tenancy`, we shouldn't need any `{namespace=...}` or `by (namespace)` in the queries.
> 
> Or am I missing something?

We also use queries as keys for our redux store. If we remove the `namespace` from query then the key would be the same for more projects, which is not good. We can look at improving that later, but for now, we need to keep `namespace` in query

**Comment by spadgett** [comment]
Needs rebase and has compiler errors:

```
ERROR in /go/src/github.com/openshift/console/frontend/public/components/dashboard/project-dashboard/utilization-card.tsx
(21,3): 'humanizeSeconds' is declared but its value is never read.

ERROR in /go/src/github.com/openshift/console/frontend/public/components/dashboard/project-dashboard/utilization-card.tsx
(22,3): 'secondsToNanoSeconds' is declared but its value is never read.

ERROR in /go/src/github.com/openshift/console/frontend/public/components/dashboard/project-dashboard/utilization-card.tsx
(23,3): 'Humanize' is declared but its value is never read.
```

File: frontend/public/components/dashboard/project-dashboard/utilization-card.tsx [hunk 4]
```diff
@@ -136,6 +200,24 @@ export const UtilizationCard = withDashboardResources(
...
+            title="Filesystem"
+            data={filesystemStats}
+            isLoading={!projectName || !filesystemUtilization}
+            humanizeValue={humanizeBinaryBytes}
+            query={queries[ProjectQueries.FILESYSTEM_USAGE]}
+            error={filesystemError}
+            TopConsumerPopover={filesystemPopover}
+          />
+          <UtilizationItem
+            title="Network"
```
Comment (andybraren) [inline frontend/public/components/dashboard/project-dashboard/utilization-card.tsx]:
```suggestion
            title="Network Transfer in/out"
```

It looks like this number is the sum of both inbound and outbound network traffic, so this label might be slightly more clear. Do you agree? Something like this seems fine until we can have two separate lines like our mockups:

image

**Pull request merged**